app/vmctl/native/client.go

```diff
@@ -179,3 +198,37 @@ func (c *Client) do(req *http.Request, expSC int) (*http.Response, error) {
 	}
 	return resp, err
 }
+
+type keyValue struct {
+	key   string
+	value string
+}
+
+func parseHeaders(headers []string) ([]keyValue, error) {
+	if len(headers) == 0 {
+		return nil, nil
+	}
+	kvs := make([]keyValue, len(headers))
+	for i, h := range headers {
+		n := strings.IndexByte(h, ':')
+		if n < 0 {
+			return nil, fmt.Errorf(`missing ':' in header %q; expecting "key: value" format`, h)
+		}
+		kv := &kvs[i]
+		kv.key = strings.TrimSpace(h[:n])
+		kv.value = strings.TrimSpace(h[n+1:])
+	}
+	return kvs, nil
+}
```

Comment:
This snippet now appears in three places in the codebase (`native/client.go`, `remoteread/remoteread.go`, `promauth/config.go`. Could be worth extracting it out as a utility function somewhere

Reply:
Yes, that functions are similar, but in the next improvements some clients would be unified